soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java

```diff
@@ -0,0 +1,298 @@
+import org.dromara.soul.common.dto.SelectorData;
+import org.dromara.soul.common.enums.MatchModeEnum;
+import org.dromara.soul.common.enums.OperatorEnum;
+import org.dromara.soul.common.enums.ParamTypeEnum;
+import org.dromara.soul.common.enums.PluginEnum;
+import org.dromara.soul.common.enums.SelectorTypeEnum;
+import org.dromara.soul.sync.data.api.AuthDataSubscriber;
+import org.dromara.soul.sync.data.api.MetaDataSubscriber;
+import org.dromara.soul.sync.data.api.PluginDataSubscriber;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Test cases for ZookeeperSyncDataService.
+ *
+ * @author zendwang
+ */
+@Slf4j
+@SuppressWarnings("all")
+public final class ZookeeperSyncDataServiceTest {
+    
+    private static final Map<String, PluginData> PLUGIN_ZK_DTO_MAP = Maps.newConcurrentMap();
+    
+    private static TestingServer zkServer;
+    
+    private ZkClient zkClient;
+    
+    private ZookeeperSyncDataService syncDataService;
+    
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        zkServer = new TestingServer(21810, true);
+    }
+    
+    @Before
+    public void setUp() throws Exception {
+        zkClient = new ZkClient("127.0.0.1:21810");
+        buildZkData();
+        PluginDataSubscriber pluginDataSubscriber = new PluginDataSubscriber() {
+            @Override
+            public void onSubscribe(final PluginData pluginData) {
+                log.info("PluginDataSubscriber.onSubscribe,PluginData:{}", pluginData);
+            }
+
+            @Override
+            public void unSubscribe(final PluginData pluginData) {
+                log.info("PluginDataSubscriber.unSubscribe,PluginData:{}", pluginData);
+            }
+        };
+        MetaDataSubscriber metaDataSubscriber = new MetaDataSubscriber() {
+            @Override
+            public void onSubscribe(final MetaData metaData) {
+                log.info("MetaDataSubscriber.onSubscribe,MetaData:{}", metaData);
+            }
+
+            @Override
+            public void unSubscribe(final MetaData metaData) {
+                log.info("MetaDataSubscriber.unSubscribe,MetaData:{}", metaData);
+            }
+        };
+        List<MetaDataSubscriber> metaDataSubscribers = Lists.newArrayList(metaDataSubscriber);
+        AuthDataSubscriber authDataSubscriber = new AuthDataSubscriber() {
+            @Override
+            public void onSubscribe(final AppAuthData appAuthData) {
+                log.info("AuthDataSubscriber.onSubscribe,AppAuthData:{}", appAuthData);
+            }
+
+            @Override
+            public void unSubscribe(final AppAuthData appAuthData) {
+                log.info("AuthDataSubscriber.unSubscribe,AppAuthData:{}", appAuthData);
+            }
+        };
```

Comment:
Hey~ I suggest you can do like. Not just logging.
```
@SneakyThrows
    @Test
    public void testWatcherPlugin() {
        final CountDownLatch latch = new CountDownLatch(1);
        PluginData pluginData = new PluginData("6", PluginEnum.DIVIDE.getName(), "aaaaa", 0, Boolean.TRUE);
        writePlugin(pluginData);
        final List<PluginData> actuals = new ArrayList<>();
        syncDataService = new ZookeeperSyncDataService(zkClient, new PluginDataSubscriber() {
            @Override
            public void onSubscribe(PluginData pluginData) {
                latch.countDown();
                actuals.add(pluginData);
            }
        }, Collections.emptyList(), Collections.emptyList());
        latch.await(3, TimeUnit.SECONDS);
        Assert.assertEquals(1, actuals.size());
        Assert.assertEquals(pluginData, actuals.get(0));
    }
```

Reply:
I will try to refactor  in this way!